src/neo4j/_async/work/result.py

```diff
@@ -242,17 +280,81 @@ def on_success(summary_metadata):
             self._metadata.update(summary_metadata)
             self._bookmark = summary_metadata.get("bookmark")
             self._database = summary_metadata.get("db", self._database)
+            await _on_summary()
+            self._handle_warnings()
 
         # This was the last page received, discard the rest
         self._connection.discard(
             n=-1,
             qid=self._qid,
             on_success=on_success,
             on_failure=on_failure,
-            on_summary=on_summary,
         )
         self._streaming = True
 
+    def _handle_warnings(self) -> None:
+        sev_filter: t.Tuple[NotificationMinimumSeverity, ...] = ()
+        if (
+            self._warn_notification_severity
+            == NotificationMinimumSeverity.WARNING
+        ):
+            sev_filter = (NotificationMinimumSeverity.WARNING,)
+        elif (
+            self._warn_notification_severity
+            == NotificationMinimumSeverity.INFORMATION
+        ):
+            sev_filter = (NotificationMinimumSeverity.INFORMATION,
+                          NotificationMinimumSeverity.WARNING)
+
+        summary = self._obtain_summary()
+        query = self._metadata.get("query")
+        for notification in summary.summary_notifications:
+            log_call = notification_log.debug
+            if notification.severity_level == NotificationSeverity.INFORMATION:
+                log_call = notification_log.info
+            elif notification.severity_level == NotificationSeverity.WARNING:
+                log_call = notification_log.warning
+            log_call(
+                "Received notification from DBMS server: %s",
+                NotificationPrinter(notification, query, one_line=True)
+            )
+
+            if notification.severity_level not in sev_filter:
+                continue
```

Comment:
Shouldn't it be filter before logging?

Reply:
I called the config option `warn_notification_severity` hoping to convey that I only intended to be filtering what notifications cause a `warning`. It wasn't meant to affect logging. If you don't care about the logging, you can always do `logging.getLogger("neo4j.notification").disabled = True` or similar.

Reply:
Resolved in a call.